pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -65,7 +65,15 @@ private AggregationFunctionUtils() {
   @Nonnull
   public static String formatValue(@Nonnull Object value) {
     if (value instanceof Double) {
-      return String.format(Locale.US, "%1.5f", (Double) value);
+      Double doubleValue = (Double) value;
+
+      // String.format is very expensive, so avoid it for whole numbers that can fit in Long.
+      // We simply append ".00000" to long, in order to keep the existing behavior.
+      if (doubleValue < Long.MAX_VALUE && !Double.isInfinite(doubleValue) && doubleValue == Math.floor(doubleValue)) {
```

Comment:
Use DoubleMath.isMathematicalInteger(value) instead, which should be faster.
https://google.github.io/guava/releases/19.0/api/docs/com/google/common/math/DoubleMath.html#isMathematicalInteger(double)